### DISCUSSION TITLE
Optimization?

### DISCUSSION THREAD

**Discussion by vbjay** [dotnet/runtime#84232] [Q&A]
Different ways to optimize but I wonder if it would it be better to instead of looping through all the colors in ColorTable each time and calling ToARGB and then checking each one against target,  would one or more of the following be better:

- Use a double key value where the key can be searched by ARGB or Name
- use a hashset of Argb values and the colors so the argb values are cached as long as ColorTable is around.
- or other idea

Yes I know the ColorTable isn't huge.  So just more of a would another way eek out anything better or is it good enough.

https://source.dot.net/#System.ComponentModel.TypeConverter/src/libraries/Common/src/System/Drawing/ColorConverterCommon.cs,608eb04e851354ba,references

**Comment by danmoseley** [comment]
If this is something that tends to appear on hot paths, and you can offer a PR that speeds it up without adding unreasonable complexity, we'd likely welcome that.

(Is it showing up in hot path for your projects?)

**Reply by vbjay** [reply]
I was tracking down how some of the color code works and noticed it. Started looking into it. Figured a discussion to see if gut was right and worth it. Not huge priority. Just a gut check and a potential improvement.
